.github/workflows/step_setup_and_build.yaml

```diff
@@ -54,6 +54,9 @@ jobs:
             echo "apiSubDomain=dev" >> $GITHUB_ENV
           fi
 
+      - name: test project
+        run: pnpm test
```

Comment:
Generally, testing is a separate pipeline. i.e. It's possible to successfully build the project, but not pass all the tests. 

A general approach for coordinating this is to build and deploy the website in one pipeline so you can get immediate feedback and preview. Then run tests, type checking, quality control in a separate pipeline because these are issues that should be addressed by the developer as they get around to it.

 i.e. A separation of concerns exists between fixing the website and fixing the tests, and making them dependent on each other can slow down both processes.